t/eth_test.t

```diff
@@ -0,0 +1,181 @@
+        type         => 'internal',
+        timestamp    => 1599542853,
+        contract     => '',
+        data         => '0x',
+    );
+
+    # $loop->add(my $blockchain_eth = Net::Async::Blockchain::ETH->new());
+
+    $mock_rpc->mock(
+        get_block_by_number => async sub {
+            return $sample_get_block_by_number;
+        },
+        get_transaction_receipt => async sub {
+            return $sample_get_transaction_receipt;
+        });
+
+    $mock_eth->mock(
+        accounts => sub {
+            my %accounts = (
+                lc "0x8295507db4b0d6a18f6c69be7d5484d5dac3ed9d" => 1,
+                lc "0x1f618fd55aaba65f1551c10b6022b7f9f0a2224c" => 1
+            );
+            return \%accounts;
+        },
+        _check_contract_transaction => async sub {
+            return ();
+        },
+        _set_transaction_type => async sub {
+            my ($self, $transaction) = @_;
+            $transaction->{type} = 'internal';
+            return $transaction;
+        },
+        latest_accounts_update => sub {
+            return time;
+        });
+
+    my $blockchain_eth_source = $blockchain_eth->source;
+    $blockchain_eth_source->each(
+        sub {
+            my $emitted_transaction = shift;
+            is_deeply $emitted_transaction, $expected_transaction, "Correct emitted trasnaction";
+            $blockchain_eth_source->finish();
+        });
+
+    $blockchain_eth->newHeads({params => {result => {number => '0x6d56f'}}})->get;
+    $blockchain_eth_source->get;
+
+    $mock_rpc->unmock_all();
+    $mock_eth->unmock_all();
+};
+
+subtest "Test Case - to check transacform_transaction WITHOUT Receipt" => sub {
+
+    my $decoded_transaction = {
+        'hash'             => '0x418aeca12f07c7010109b03d5d012f3c93f922ab520e831efb29075ae4903489',
+        'to'               => '0x8295507db4b0d6a18f6c69be7d5484d5dac3ed9d',
+        'nonce'            => '0x56',
+        'gas'              => '0x5208',
+        'v'                => '0xa95',
+        'blockNumber'      => '0x6d56f',
+        'value'            => '0xa688906bd8b00000',
+        'transactionIndex' => '0x0',
+        'gasPrice'         => '0x1',
+        'r'                => '0x4f272382dafff68bd7167bf4726397bef5df1d039ebdc16286833f40bb0d4b86',
+        'input'            => '0x',
+        'blockHash'        => '0x1285db1573a082bbad24475599762f9460eccc49868884a944527605efede14d',
+        's'                => '0x403dc625a9bf3f062bc6d721bb02c8e57bb353917e719e54a49929deb438d56f',
+        'from'             => '0x1f618fd55aaba65f1551c10b6022b7f9f0a2224c'
+    };
+
+    $mock_rpc->mock(
+        get_transaction_receipt => async sub {
+            return undef;
+        });
+
+    my $response = $blockchain_eth->transform_transaction($decoded_transaction, $sample_get_block_by_number->{timestamp})->get;
+    is $response, 0, "Correct response for transaction without receipt";
+
+    $mock_rpc->unmock_all();
+};
+
+done_testing;
```

Comment:
IMO, better to improve the test with reading and pushing to/from Redis to make sure you are processing the correct transaction